test/parallel/test-fs-mkdtemp.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+common.refreshTmpDir();
+
+fs.mkdtempSync(common.tmpDir + '/foo.XXXXXX');
+assert(fs.readdirSync(common.tmpDir).some(function(folder) {
+    return folder.indexOf('foo.') === 0 && folder.length === 'foo.XXXXXX'.length;
+}));
+
+fs.mkdtemp(common.tmpDir + '/bar.XXXXXX', function(err, folder) {
+    assert(common.fileExists(folder));
+});
+
+fs.mkdtemp(common.tmpDir + '/bar', function(err) {
+    assert(err !== null);
```

Comment:
It seems libuv handles `EINVAL` properly: https://github.com/libuv/libuv/blob/v1.x/src/win/fs.c#L783

I'm not sure whether:

- The returned error object in node userland is actually `{code: 'EINVAL'}`
- The other errors (e.g. "No space left on device") are actually handled

The 2nd case probably doesn't matter though, so I could just check `EINVAL` and the test would work everwhere.